azext_edge/edge/util/file_operations.py

```diff
@@ -99,38 +99,35 @@ def deserialize_file_content(file_path: str) -> Any:
     if not valid_extension or extension == "json":
         # will always be a list or dict
         result = _try_loading_as(
-            loader=json.loads,
-            content=content,
-            error_type=json.JSONDecodeError,
-            raise_error=valid_extension
+            loader=json.loads, content=content, error_type=json.JSONDecodeError, raise_error=valid_extension
         )
     if (not result and not valid_extension) or extension in ["yaml", "yml"]:
         # can be list, dict, str, int, bool, none
         result = _try_loading_as(
-            loader=yaml.safe_load,
-            content=content,
-            error_type=yaml.YAMLError,
-            raise_error=valid_extension
+            loader=yaml.safe_load, content=content, error_type=yaml.YAMLError, raise_error=valid_extension
         )
     if (not result and not valid_extension) or extension == "csv":
         # iterrable object so lets cast to list
         result = _try_loading_as(
-            loader=csv.DictReader,
-            content=content.splitlines(),
-            error_type=csv.Error,
-            raise_error=valid_extension
+            loader=csv.DictReader, content=content.splitlines(), error_type=csv.Error, raise_error=valid_extension
         )
     if result is not None or valid_extension:
         return result
     raise FileOperationError(f"File contents for {file_path} cannot be read.")
 
 
-def _try_loading_as(
-    loader: Callable,
-    content: str,
-    error_type: Exception,
-    raise_error: bool = True
-) -> Optional[Any]:
+def validate_file_extension(file_name: str, expected_exts: List[str]) -> str:
+    ext = os.path.splitext(file_name)[1]
+    if ext not in expected_exts:
```

Comment:
I think this should be a case-insensitive comparison

Reply:
sounds good!